packages/@aws-cdk/aws-servicecatalogappregistry/lib/application.ts

```diff
@@ -12,6 +12,32 @@ import { CfnApplication, CfnAttributeGroupAssociation, CfnResourceAssociation }
 const APPLICATION_READ_ONLY_RAM_PERMISSION_ARN = 'arn:aws:ram::aws:permission/AWSRAMPermissionServiceCatalogAppRegistryApplicationReadOnly';
 const APPLICATION_ALLOW_ACCESS_RAM_PERMISSION_ARN = 'arn:aws:ram::aws:permission/AWSRAMPermissionServiceCatalogAppRegistryApplicationAllowAssociation';
 
+/**
+ * Properties for a Service Catalog AppRegistry Attribute Group
+ */
+export interface AttributeGroupAssociationProps {
+  /**
+   * Logical Id of the attribute group to be created
+   */
+  readonly attributeGroupLogicalId: string;
+  /**
+   * Enforces a particular physical attribute group name.
+   */
+  readonly attributeGroupName: string;
```

Comment:
Why is this not optional?

Reply:
Attribute group name is the key to calculate attribute group construct id to make sure that multiple `addAttributeGroup` calls won't generate the same construct ID. Since name is unique, if `addAttributeGroup` is called twice with the same `attributeGroupName`, construct duplicate error will be thrown. Both `Names.uniqueNodeId()` and `Names.uniqueResourceName()` generate id from the path of the construct, i.e. path of the application. As a result,  two `addAttributeGroup` calls with different names\attributes\descriptions generate two `AttributeGroup` constructs with the same construct ID.

Reply:
This should be required even if we use customer provided construct id for attribute group.

Attribute group name is the key human readable identifier our service offers for customer to find the attribute group, we want to make it explicit, so that it will can make use of the name to look for the attribute group in other platforms like console or SDK client.